src/dodal/devices/fast_grid_scan.py

```diff
@@ -181,181 +196,165 @@ def non_integer_dwell_time(cls, dwell_time_ms: float) -> float:
+
+        self.movable_params: dict[str, Signal] = {
+            "x_steps": self.x_steps,
+            "y_steps": self.y_steps,
+            "z_steps": self.z_steps,
+            "x_step_size": self.x_step_size,
+            "y_step_size": self.y_step_size,
+            "z_step_size": self.z_step_size,
+            "x_start": self.x_start,
+            "y1_start": self.y1_start,
+            "y2_start": self.y2_start,
+            "z1_start": self.z1_start,
+            "z2_start": self.z2_start,
+        }
+
+    @AsyncStatus.wrap
+    async def kickoff(self):
+        curr_prog = await self.motion_program.program_number.get_value()
+        running = await self.motion_program.running.get_value()
+        if running:
+            LOGGER.info(f"Motion program {curr_prog} still running, waiting...")
+            await wait_for_value(self.motion_program.running, 0, self.KICKOFF_TIMEOUT)
+
+        LOGGER.debug("Running scan")
+        await self.run_cmd.set(1)
+        LOGGER.info("Waiting for FGS to start")
+        await wait_for_value(self.status, 1, self.KICKOFF_TIMEOUT)
+        LOGGER.debug("FGS kicked off")
+
+    @AsyncStatus.wrap
+    async def complete(self):
+        await wait_for_value(self.status, 0, self.COMPLETE_STATUS)
+
+
+class ZebraFastGridScan(FastGridScanCommon[ZebraGridScanParams]):
+    """Device for standard Zebra FGS. In this scan, the goniometer's velocity profile follows a parabolic shape between X steps,
+    with the slowest points occuring at each X step.
+    """
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        super().__init__(prefix, name)
+
+        # Time taken to travel between X steps
+        self.dwell_time_ms = epics_signal_rw_rbv(float, "DWELL_TIME")
+        self.movable_params["dwell_time_ms"] = self.dwell_time_ms
+
+
+class PandAFastGridScan(FastGridScanCommon[PandAGridScanParams]):
+    """Device for panda constant-motion scan"""
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        super().__init__(prefix, name)
+        self.time_between_x_steps_ms = (
+            epics_signal_rw_rbv(  # Used by motion controller to set goniometer velocity
+                float, "TIME_BETWEEN_X_STEPS"
             )
+        )
 
-    def _running_changed(self, value=None, old_value=None, **kwargs):
-        if (old_value == 1) and (value == 0):
-            self.set_finished()
-            self.clean_up()
+        # Distance before and after the grid given to allow goniometer to reach desired speed while it is within the
+        # grid
+        self.run_up_distance_mm = epics_signal_rw_rbv(float, "RUNUP_DISTANCE")
+        self.movable_params["run_up_distance_mm"] = self.run_up_distance_mm
 
-    def clean_up(self):
-        self.device.position_counter.clear_sub(self._notify_watchers)
-        self.device.status.clear_sub(self._running_changed)
 
+def set_fast_grid_scan_params(
+    scan: FastGridScanCommon[ParamType], params: GridScanParamsCommon
```

Comment:
Must: `params` should be of type `ParamType`. This means we can then remove the assert and just trust the type system to catch when the params don't match